### PR TITLE
11 feature 5 assign observer

### PR DESCRIPTION
**Pull request by Nicknames43**
NOTE: When it says there are two participants, that does not mean that the other group member (Aditya) wrote code for this feature - rather I pulled code into this branch that they wrote for a _different feature_. This can be seen in the commit history.
NOTE: The errors in the auto-grade seem to be coming from the fact that I did not commit and push anything from the generalClasses package despite the code using it since it can be found in a different branch that currently has a pull request. Once that branch has been merged to main I will merge with main and update the pull request if needed.

This is a pull request for the feature relating to allowing the overseer user to assign an observer to a particular game. This was designed using the principles of clean architecture. This pull request contains the following new classes: AssignObserverUC (the use case interactor), AssignObserverID (the input data/request model), AssignObserverOD (output data/response model), AssignObserverIB (input boundary), and AssignObserverOB (output boundary). 

The input data takes the assignee and the gameID of the game the observer is being assigned to. The interactor implements the input boundary and does the following: provided all necessary conditions are met, the assignee becomes the observer responsible for the chosen game. The output data packages and brings data out to the presenter.

### CODE REVIEW

**Review by yibinglu — APPROVED** [review]
Overall logic looks good for this feature, everything seems to follow clean architecture. The one thing I'm a little concerned about is not directly related to your code, but it says that this PR is failing the GitHub auto grading so there's a small red "x" next to your pull request instead of the green checkmark. I'm not sure if that means that we can't merge it but I would maybe look into that and see what might be causing it. Other than that, everything looks good so I'll go ahead and approve it.

**Review by Kelleou — APPROVED** [review]
Your code follows clean architecture and I think you only need to implement the controller and presenter.
My only concern is the TreeMethods and the PermissionChecker class that is used in the use case because they are imported in AssignObserverUC but they are not in the commits nor in main (yet in Aditya's pull request, I was able to see TreeMethods and PermissionChecker in his files) so I think this might be the reason why your pull request did not pass the auto grading in github. But once we merge Aditya's branch, the generalClasses package will be in main so we should be fine.

**Review by AdK0101 — APPROVED** [review]
I agree with the other reviewers. There is not really anything wrong with your code as is (provided we merge with the helper methods that I defined). And I already have a general level of familiarity with your code since we have discussed it beforehand. So if I see something wrong with it in the future, I will let you know.

**Pull request closed without merging**
